core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -167,8 +173,25 @@ SYCL::SYCLDevice::SYCLDevice(size_t id) {
 
 sycl::device SYCL::SYCLDevice::get_device() const { return m_device; }
 
-void SYCL::impl_initialize(SYCL::SYCLDevice d) {
-  Impl::SYCLInternal::singleton().initialize(d.get_device());
+void SYCL::impl_initialize(InitArguments const& args) {
+  // If there are no GPUs return whatever else we can run on if no specific GPU
+  // is requested.
+  const auto num_gpus =
+      sycl::device::get_devices(sycl::info::device_type::gpu).size();
+  int use_gpu = num_gpus == 0 ? args.device_id : Kokkos::Impl::get_gpu(args);
+
+  if (std::is_same<Kokkos::Experimental::SYCL,
+                   Kokkos::DefaultExecutionSpace>::value ||
```

Comment:
This is always true isn't it?

Reply:
You could use
```
#define KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_THREADS
```
or similar to change the default execution space.